--- managingnews-1.0-beta8/profiles/managingnews/modules/contrib/feeds/libraries/common_syndication_parser.inc	2010-01-28 20:57:54.000000000 +0100
+++ mn_beta8/profiles/managingnews/modules/contrib/feeds/libraries/common_syndication_parser.inc	2010-03-24 12:37:10.000000000 +0100
@@ -260,16 +260,15 @@ function _parser_common_syndication_RDF1
     }
 
     // Declaratively define mappings that determine how to construct the result object.
-    $item = _parser_common_syndication_RDF10_item($rdf_data, (object)array(
+    $item = _parser_common_syndication_RDF10_item($rdf_data, array(
       'title'       => array('rss:title', 'dc:title'),
       'description' => array('rss:description', 'dc:description', 'content:encoded'),
-      'options'     => (object)array(
-        'guid'            => 'rdf:about',
-        'timestamp'       => 'dc:date',
-        'author' => array('dc:creator', 'dc:publisher'),
-        'url'    => array('rss:link', 'rdf:about'),
-        'tags'            => 'dc:subject',
-      ),
+      'url'         => array('rss:link', 'rdf:about'),
+      'author'      => array('dc:creator', 'dc:publisher'),
+      'guid'        => 'rdf:about',
+      'timestamp'   => 'dc:date',
+      'author'      => array('dc:creator', 'dc:publisher'),  
+      'tags'        => 'dc:subject'
     ));
 
     // Special handling for the title:
@@ -289,7 +288,7 @@ function _parser_common_syndication_RDF1
     }
 
     // Add every found RDF property to the feed item.
-    $item['rdf'] = (object)array();
+    $item['rdf'] = array();
     foreach ($rdf_data as $rdf_property => $rdf_value) {
       // looks nicer in the mapper UI
       // @todo Revisit, not used with feedapi mapper anymore.
@@ -299,7 +298,7 @@ function _parser_common_syndication_RDF1
 
     $parsed_source['items'][] = $item;
   }
-
+ 
   return $parsed_source;
 }
 
@@ -314,16 +313,11 @@ function _parser_common_syndication_RDF1
 }
 
 function _parser_common_syndication_RDF10_item($rdf_data, $mappings) {
-  foreach (get_object_vars($mappings) as $k => $v) {
-    if (is_object($v)) {
-      $mappings->$k = _parser_common_syndication_RDF10_item($rdf_data, $v);
-    }
-    else {
-      $values = _parser_common_syndication_RDF10_property($rdf_data, $v);
-      $mappings->$k = !is_array($values) || count($values) > 1 ? $values : reset($values);
-    }
+  foreach ( $mappings as $k => $v) {
+    $values = _parser_common_syndication_RDF10_property($rdf_data, $v);
+    $mappings[$k] = !is_array($values) || count($values) > 1 ? $values : reset($values);
   }
-  return (object)$mappings;
+  return $mappings;
 }
 
 /**
