Index: outline.module
===================================================================
--- outline.module	(revision 46)
+++ outline.module	(working copy)
@@ -939,6 +939,12 @@
   if (user_access('administer book outlines')) {
     $outline_action = TRUE;
   }
+  elseif ($node->uid == $user->uid) {
+    $outline_action = TRUE;
+  }
+  elseif (  array_key_exists($user->uid, $node->coherent_access['editors']) ) {
+    $outline_action = TRUE;
+  }
   elseif ($node->outline['books'][$bid]['book_role_perm']) {
     $rids = array();
     foreach ($user->roles AS $rid => $role) {
